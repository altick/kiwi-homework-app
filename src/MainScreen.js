// @flow

import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import MainContext from './MainContext';
import { Container, Header, Left, Body, Title, Segment, Right, Content, Footer, FooterTab, Button, Item, Input, Grid, Col, Row } from 'native-base';

const MODE_PREDICT = 'predict';
const MODE_EXPAND = 'expand';

class MainScreen extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            mode: MODE_PREDICT,
            input: '',
            numStr: '',
            expansions: [],
            selectedExpansion: ''
        };
    }

    async componentDidMount() {
        // this.props.actions.someAction('hello', 'World').then(() => {
        //     // this.props.navigation.push('Next Screen');
        // });
    }
    
    componentDidUpdate(prevProps, prevState) {
        // Previous SomeContext value is prevProps.someValue
        // New SomeContext value is this.props.someValue
    }

    async updateExpansions() {
        const { numStr } = this.state;
        let expansions = await this.props.actions.getExpansions(numStr);
        console.log(expansions);

        this.setState({
            selectedExpansion: expansions.length ? expansions[0] : '',
            expansions: expansions
        });
    }

    setMode(mode) {
        this.setState({ mode: mode });
    }

    appendInput(text) {
        const { input } = this.state;
        this.setState({
            input: input + text
        });
    }

    removeLastLetter() {
        const { input } = this.state;
        if(input.length == 0) {
            return;
        }
        this.setState({
            input: input.substring(0, input.length - 1)
        });
    }

    appendSpace() {
        const { numStr, selectedExpansion } = this.state;
        this.appendInput(selectedExpansion + ' ');

        this.setState({
            numStr: ''
        });
    }

    async appendNumbericString(number) {
        const { numStr } = this.state;
        this.setState({
            numStr: numStr + number
        });
        await this.updateExpansions();
    }

    async removeLastNumber() {
        const { numStr } = this.state;
        if(numStr.length == 0) {
            return;
        }
        this.setState({
            numStr: numStr.substring(0, numStr.length - 1)
        });
        await this.updateExpansions();
    }

    onKeyClick(number) {
        const { numStr } = this.state;
        switch(number) {
            case '0': {
                this.appendSpace();
                break;
            }
            case '1': {
                if(numStr.length) {
                    this.removeLastNumber();
                } else {
                    this.removeLastLetter();
                }
                break;
            }
            case '2':
            case '3':
            case '4':
            case '5':
            case '6':
            case '7':
            case '8':
            case '9':
            {
                this.appendNumbericString(number);
                break;
            }
            default: {
                // do nothing
            }
        }
    }

    renderButton(number, letters) {
        return (
            <Button full large rounded light style={ styles.keyboardButton } onPress={ () => this.onKeyClick(number) }>
                <View>
                    <Text style={ styles.keyNumber }>{ number }</Text>
                    <Text style={ styles.keyLetters }>{ letters }</Text>
                </View>
            </Button>
        );
    }

    render() {
        // const { someValue, actions: { someAction } } = this.props;
        const { mode, input, numStr, expansions, selectedExpansion } = this.state;
        
        return (
            <Container>
                <Header hasSegment>
                    <Left />
                    <Body>
                        <Title>Kiwi T9 App</Title>
                    </Body>
                    <Right />
                </Header>
                <Segment>
                    <Button first active={ mode == MODE_PREDICT } onPress={ () => this.setMode(MODE_PREDICT) }><Text>Predictions</Text></Button>
                    <Button last active={ mode == MODE_EXPAND } onPress={ () => this.setMode(MODE_EXPAND) }><Text>Expansions</Text></Button>
                </Segment>
                <Content>
                    <Grid>
                        <Row>
                            <Col>
                                <Item full rounded>
                                    <Input placeholder='Input' value={ input + numStr + '_' }/>
                                </Item>
                            </Col>
                        </Row>
                        <Row>
                            { expansions.length > 0 
                                ? expansions.map((expansion, i) => (
                                    <Col key={i}>
                                        <Button full>
                                            <Text style={ expansion == selectedExpansion ? styles.selectedExpansion : styles.expansion }>{ expansion }</Text>
                                        </Button>
                                    </Col>
                                )) 
                                :  <Col><Button full><Text>...</Text></Button></Col>
                            }
                        </Row>
                        <Row>
                            <Col>{ this.renderButton('1', '< C')}</Col>
                            <Col>{ this.renderButton('2', 'abc')}</Col>
                            <Col>{ this.renderButton('3', 'def')}</Col>
                        </Row>
                        <Row>
                            <Col>{ this.renderButton('4', 'ghi')}</Col>
                            <Col>{ this.renderButton('5', 'jkl')}</Col>
                            <Col>{ this.renderButton('6', 'mno')}</Col>
                        </Row>
                        <Row>
                            <Col>{ this.renderButton('7', 'pqrs')}</Col>
                            <Col>{ this.renderButton('8', 'tuv')}</Col>
                            <Col>{ this.renderButton('9', 'wxyz')}</Col>
                        </Row>
                        <Row>
                            <Col>{ this.renderButton('#', '')}</Col>
                            <Col>{ this.renderButton('0', '_')}</Col>
                            <Col>{ this.renderButton('*', '')}</Col>
                        </Row>
                    </Grid>
                </Content>
                <Footer>
                    <FooterTab>
                        <Button full>
                            <Text>Footer</Text>
                        </Button>
                    </FooterTab>
                </Footer>
            </Container>
        );
    }
}

let styles = StyleSheet.create({
    keyboardButton: {
        padding: 20,
        margin: 4
    },
    keyNumber: {
        fontSize: 20,
        textAlign: 'center'
    },
    keyLetters: {
        fontSize: 12,
        textAlign: 'center'
    },
    expansion: {

    },
    selectedExpansion: {
        fontWeight: 'bold'
    }


});

let MainScreenWithContext = props => (
    <MainContext.Consumer>
        {state => (
            <MainScreen { ...props } {...state} ></MainScreen>
        ) }
    </MainContext.Consumer>
);

export default MainScreenWithContext;