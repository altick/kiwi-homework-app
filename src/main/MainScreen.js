// @flow

import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import MainContext from './MainContext';
import { Container, Header, Left, Body, Title, Segment, Right, Content, Footer, FooterTab, Button, Item, Grid, Col, Row, Textarea, Spinner, Icon } from 'native-base';

const MODE_PREDICT = 'predict';
const MODE_EXPAND = 'expand';

class MainScreen extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            mode: MODE_PREDICT,
            input: '',
            numStr: '',
            expansions: [],
            selectedExpansion: ''
        };
    }

    async componentDidMount() {
        // this.props.actions.someAction('hello', 'World').then(() => {
        //     // this.props.navigation.push('Next Screen');
        // });
    }
    
    componentDidUpdate(prevProps, prevState) {
        // Previous SomeContext value is prevProps.someValue
        // New SomeContext value is this.props.someValue
    }

    async updateExpansions(numStr) {
        let expansions = [];
        
        switch(this.state.mode) {
            case MODE_EXPAND:
            expansions = await this.props.actions.getExpansions(numStr);
            break;
            case MODE_PREDICT:
            expansions = await this.props.actions.getPredictions(numStr);
            break;
        }
        
        console.log(expansions);

        this.setState({
            selectedExpansion: expansions.length ? expansions[0] : '',
            expansions: expansions
        });
    }

    setMode(mode) {
        const { numStr } = this.state;

        this.setState({ mode: mode });

        this.updateExpansions(numStr);
    }

    clearExpansions() {
        this.setState({
            numStr: '',
            expansions: [],
            selectedExpansion: '',
        });
    }

    applySelectedExpansion() {
        const { selectedExpansion } = this.state;
        this.applyExpansion(selectedExpansion);
    }

    applyExpansion(expansion) {
        this.appendInput(expansion + ' ');

        this.clearExpansions();
    }

    appendInput(text) {
        const { input } = this.state;
        this.setState({
            input: input + text
        });
    }

    removeLastLetter() {
        const { input } = this.state;
        if(input.length == 0) {
            return;
        }

        this.clearExpansions();
        this.setState({
            input: input.substring(0, input.length - 1)
        });
    }

    appendSpace() {
        const { selectedExpansion } = this.state;
        this.appendInput(selectedExpansion + ' ');

        this.clearExpansions();
    }

    async appendNumbericString(number) {
        let { numStr } = this.state;
        numStr = numStr + number;
        this.setState({
            numStr: numStr
        });
        await this.updateExpansions(numStr);
    }

    async removeLastNumber() {
        let { numStr } = this.state;
        if(numStr.length == 0) {
            return;
        }
        numStr = numStr.substring(0, numStr.length - 1);

        this.setState({
            numStr: numStr
        });
        await this.updateExpansions(numStr);
    }

    onKeyClick(number) {
        const { numStr } = this.state;
        switch(number) {
            case '0': {
                this.appendSpace();
                break;
            }
            case '1': {
                if(numStr.length) {
                    this.removeLastNumber();
                } else {
                    this.removeLastLetter();
                }
                break;
            }
            case '2':
            case '3':
            case '4':
            case '5':
            case '6':
            case '7':
            case '8':
            case '9':
            {
                this.appendNumbericString(number);
                break;
            }
            case '#':
            {
                this.selectPrevExpansion();
                break;
            }
            case '*':
            {
                this.selectNextExpansion();
                break;
            }
            default: {
                // do nothing
            }
        }
    }

    selectNextExpansion() {
        const { expansions, selectedExpansion } = this.state;

        let i = expansions.indexOf(selectedExpansion);
        if(i == -1 || i == expansions.length - 1) {
            return;
        }

        this.setState({
            selectedExpansion: expansions[i + 1]
        });
    }

    selectPrevExpansion() {
        const { expansions, selectedExpansion } = this.state;

        let i = expansions.indexOf(selectedExpansion);
        if(i == -1 || i == 0) {
            return;
        }

        this.setState({
            selectedExpansion: expansions[i - 1]
        });
    }

    onExpansionClick(index) {
        let expansion = this.state.expansions[index];

        this.applyExpansion(expansion);
    }

    renderButton(number, letters) {
        return (
            <Button full large rounded light style={ styles.keyboardButton } onPress={ () => this.onKeyClick(number) }>
                <View>
                    <Text style={ styles.keyNumber }>{ number }</Text>
                    <Text style={ styles.keyLetters }>{ letters }</Text>
                </View>
            </Button>
        );
    }

    renderExpansions() {
        const { expansions, selectedExpansion } = this.state;

        if(expansions.length == 0) {
            return (
                <Col style={ styles.expansionCotainer }><Button light full><Text>...</Text></Button></Col>
            );
        } else {
            return expansions.map((expansion, i) => (
                <Col key={i} style={ styles.expansionCotainer }>
                    <Button light full onPress={ () => this.onExpansionClick(i) }>
                        <Text style={ expansion == selectedExpansion ? styles.selectedExpansionText : styles.expansionText }>{ expansion }</Text>
                    </Button>
                </Col>
            ));
        }
    }

    render() {
        // const { someValue, actions: { someAction } } = this.props;
        const { mode, input, selectedExpansion } = this.state;
        const { isLoading } = this.props;

        return (
            <Container>
                <Header hasSegment>
                    <Left />
                    <Body>
                        <Title>Kiwi T9 App</Title>
                    </Body>
                    <Right>
                        <Button transparent>
                            <Icon name="md-settings" type="Ionicons" />
                        </Button>
                    </Right>
                </Header>
                <Segment>
                    <Button first active={ mode == MODE_PREDICT } onPress={ () => this.setMode(MODE_PREDICT) }><Text>Predictions</Text></Button>
                    <Button last active={ mode == MODE_EXPAND } onPress={ () => this.setMode(MODE_EXPAND) }><Text>Expansions</Text></Button>
                </Segment>
                <Grid>
                    <Row style={ { flex: 2 } }>
                        <Col>
                            <Item full style={ styles.textArea }   style={ { flex: 1 } } >
                                <Textarea rowSpan={3} value={ input + selectedExpansion + '_' }  style={ { flex: 1 } } />
                            </Item>
                        </Col>
                    </Row>
                    <Row>
                        { this.renderExpansions() }
                        { isLoading && <View style={ styles.expansionsLoader }><Spinner /></View> }
                    </Row>
                    <Row>
                        <Col>{ this.renderButton('1', '< C')}</Col>
                        <Col>{ this.renderButton('2', 'abc')}</Col>
                        <Col>{ this.renderButton('3', 'def')}</Col>
                    </Row>
                    <Row>
                        <Col>{ this.renderButton('4', 'ghi')}</Col>
                        <Col>{ this.renderButton('5', 'jkl')}</Col>
                        <Col>{ this.renderButton('6', 'mno')}</Col>
                    </Row>
                    <Row>
                        <Col>{ this.renderButton('7', 'pqrs')}</Col>
                        <Col>{ this.renderButton('8', 'tuv')}</Col>
                        <Col>{ this.renderButton('9', 'wxyz')}</Col>
                    </Row>
                    <Row>
                        <Col>{ this.renderButton('#', '< prev')}</Col>
                        <Col>{ this.renderButton('0', '_')}</Col>
                        <Col>{ this.renderButton('*', 'next >')}</Col>
                    </Row>
                </Grid>
            </Container>
        );
    }
}

let styles = StyleSheet.create({
    textArea: {
        padding: 5
    },
    keyboardButton: {
        padding: 20,
        margin: 4
    },
    keyNumber: {
        fontSize: 20,
        textAlign: 'center'
    },
    keyLetters: {
        fontSize: 12,
        textAlign: 'center'
    },
    expansionCotainer: {
        zIndex: 1
    },
    expansionText: {

    },
    selectedExpansionText: {
        fontWeight: 'bold'
    },
    expansionsLoader: { 
        position: 'absolute', 
        left: 0, 
        top: -17, 
        zIndex: 10 
    }
});

let MainScreenWithContext = props => (
    <MainContext.Consumer>
        {state => (
            <MainScreen { ...props } {...state} ></MainScreen>
        ) }
    </MainContext.Consumer>
);

export default MainScreenWithContext;