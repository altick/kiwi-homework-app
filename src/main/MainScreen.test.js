import React from 'react';
import renderer from 'react-test-renderer';
import { MainScreen, MODE_EXPAND, MODE_PREDICT } from './MainScreen';

let rendered: renderer.ReactTestRenderer = null;
let instance: MainScreen = null;

beforeEach(() => {
    rendered = renderer.create(<MainScreen />);
    instance = rendered.getInstance();
})

describe('Input mode', () => {

    describe('Set Mode', () => {

        it('Should set EXPAND mode', () => {
            instance.setMode(MODE_EXPAND);
            expect(instance.state.mode).toBe(MODE_EXPAND);
        });

        it('Should set PREDICT mode', () => {
            instance.setMode(MODE_PREDICT);
            expect(instance.state.mode).toBe(MODE_PREDICT);
        });

    });

    describe('Set Mode - On click', () => {

        beforeEach(() => {
            instance.setMode = jest.fn();
            instance.updateExpansions = jest.fn();
        });

        it('Should set mode when clicked on set mode', () => {
            instance.onSetModeClick(MODE_PREDICT);
    
            expect(instance.setMode).toBeCalled();
        });
    
        it('Should update expansions when clicked on set mode', () => {
            instance.onSetModeClick(MODE_PREDICT);
    
            expect(instance.updateExpansions).toBeCalled();
        });

    });

});

describe('Numeric string', () => {

    beforeEach(() => {
        instance.updateExpansions = jest.fn();
    });

    describe('Appending number to numeric string', () => {

        it('Should contain single number after appending number at inital state', async () => {
            expect.assertions(1);
    
            await instance.appendNumbericString('8');
    
            expect(instance.state.numStr).toBe('8');
        });
    
        it('Should append number to current numeric string', async () => {
            expect.assertions(1);
    
            instance.state.numStr = '123';
    
            await instance.appendNumbericString('8');
    
            expect(instance.state.numStr).toBe('1238');
        });
    
        it('Should update expansions after appending number', async () => {
            expect.assertions(1);
    
            await instance.appendNumbericString('8');
    
            expect(instance.updateExpansions).toBeCalled();
        });

    });

    describe('Removing last number from numeric string', () => {

        it('Should remove the last number', async () => {
            expect.assertions(1);
    
            instance.state.numStr = '123';
    
            await instance.removeLastNumber();
    
            expect(instance.state.numStr).toBe('12');
        });

        it('Should keep empty numeric string when numeric string is empty', async () => {
            expect.assertions(1);
    
            instance.state.numStr = '';
    
            await instance.removeLastNumber();
    
            expect(instance.state.numStr).toBe('');
        });

        it('Should update expansions after removing last number', async () => {
            expect.assertions(1);
    
            instance.state.numStr = '123';

            await instance.removeLastNumber();
    
            expect(instance.updateExpansions).toBeCalled();
        });
    });

});

describe('Input string', () => {

    beforeEach(() => {
        instance.updateExpansions = jest.fn();
        instance.clearExpansions = jest.fn();
    });

    describe('Appending string to input', () => {

        it('Should append text to the empty input', () => {
            instance.state.input = '';

            instance.appendInput('hello');

            expect(instance.state.input).toBe('hello');
        });

        it('Should append text to current input', () => {
            instance.state.input = 'hello ';

            instance.appendInput('world');

            expect(instance.state.input).toBe('hello world');
        });

    });

    describe('Removing last letter', () => {

        it('Should remove last letter from the input', () => {
            instance.state.input = 'hello';

            instance.removeLastLetter();

            expect(instance.state.input).toBe('hell');
        });

        it('Should keep empty input when input is empty', () => {
            instance.state.input = '';

            instance.removeLastLetter();

            expect(instance.state.input).toBe('');
        });

        it('Should clear expansions after removing last letter', () => {
            instance.state.input = 'hello';

            instance.removeLastLetter();
    
            expect(instance.clearExpansions).toBeCalled();
        });

    });
});

describe('Expansions', () => {

    describe('Clear Expansions', () => {
        
        it('Should clear numeric string', () => {
            instance.state = {
                numStr: 'abc'
            };
            
            instance.clearExpansions();

            expect(instance.state.numStr).toBe('');
        });

        it('Should clear expansions list and selected expansion', () => {
            instance.state = {
                expansions: ['hello', 'world'],
                selectedExpansion: 'world'
            };
            
            instance.clearExpansions();

            expect(instance.state.expansions).toHaveLength(0);
            expect(instance.state.selectedExpansion).toBe('');
        });

    });

    describe('Apply expansion', () => {

        beforeEach(() => {
            instance.clearExpansions = jest.fn();
        });

        it('Should append expansion to the input and add trailing space', () => {
            instance.state = {
                input: 'Hello '
            };

            instance.applyExpansion('world');

            expect(instance.state.input).toBe('Hello world ');
        });

        it('Should append selected expansion to the input and add trailing space', () => {
            instance.state = {
                input: 'Hello ',
                selectedExpansion: 'world'
            };

            instance.applySelectedExpansion();

            expect(instance.state.input).toBe('Hello world ');
        });

        it('Should clear expansions after applying expansion', () => {
            instance.applyExpansion('world');

            expect(instance.clearExpansions).toBeCalled();
        });

    });

    describe('Update expansions', () => {

        beforeEach(() => {
            instance.props = {
                actions: {
                    getExpansions: jest.fn( async () => ['aa', 'ab', 'ac'] ),
                    getPredictions: jest.fn(async () => ['hello', 'world'] )
                }
            };
        });

        it('Should return expansions when mode is to set to MODE_EXPAND', async () => {
            expect.assertions(2);

            instance.state = {
                mode: MODE_EXPAND
            };

            await instance.updateExpansions('1');

            expect(instance.state.expansions).toEqual(expect.arrayContaining([ 'aa', 'ab', 'ac' ]));
            expect(instance.state.expansions).toHaveLength(3);
        });

        it('Should return predictions when mode is to set to MODE_PREDICT', async () => {
            expect.assertions(2);

            instance.state = {
                mode: MODE_PREDICT
            };

            await instance.updateExpansions('1');

            expect(instance.state.expansions).toEqual(expect.arrayContaining([ 'hello', 'world' ]));
            expect(instance.state.expansions).toHaveLength(2);
        });

    });

});

describe('Keyboard', () => {

    it('Should append numeric string for keys 23456789', () => {
        let mockFn = jest.fn();
        instance.appendNumbericString = mockFn;

        instance.onKeyClick('2');
        instance.onKeyClick('3');
        instance.onKeyClick('4');
        instance.onKeyClick('5');
        instance.onKeyClick('6');
        instance.onKeyClick('7');
        instance.onKeyClick('8');
        instance.onKeyClick('9');

        expect(mockFn.mock.calls).toEqual([['2'], ['3'], ['4'], ['5'], ['6'], ['7'], ['8'], ['9']]);
    });

    it('Should apply selected expansion for key 0', () => {
        let mockFn = jest.fn();
        instance.applySelectedExpansion = mockFn;

        instance.onKeyClick('0');

        expect(mockFn).toBeCalled();
    });

    it('Should remove last number for key 1 if numeric string is not empty', () => {
        let mockFn = jest.fn();
        instance.removeLastNumber = mockFn;
        instance.state.numStr = '1';

        instance.onKeyClick('1');

        expect(mockFn).toBeCalled();
    });

    it('Should remove last letter for key 1 if numeric string is empty', () => {
        let mockFn = jest.fn();
        instance.removeLastLetter = mockFn;
        instance.state.numStr = '';

        instance.onKeyClick('1');

        expect(mockFn).toBeCalled();
    });

    it('Should select previous expansion for key *', () => {
        let mockFn = jest.fn();
        instance.selectPrevExpansion = mockFn;

        instance.onKeyClick('*');

        expect(mockFn).toBeCalled();
    });

    it('Should select next expansion for key #', () => {
        let mockFn = jest.fn();
        instance.selectNextExpansion = mockFn;

        instance.onKeyClick('#');

        expect(mockFn).toBeCalled();
    });
});
