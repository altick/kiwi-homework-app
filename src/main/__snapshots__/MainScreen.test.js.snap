// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Render button Should render button with correct number and letters 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#f4f4f4",
      "borderBottomWidth": null,
      "borderColor": "#007aff",
      "borderLeftWidth": null,
      "borderRadius": 57,
      "borderRightWidth": null,
      "borderTopWidth": null,
      "borderWidth": undefined,
      "elevation": 2,
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "center",
      "margin": 4,
      "opacity": 1,
      "padding": 20,
      "paddingBottom": 6,
      "paddingTop": 6,
      "shadowColor": undefined,
      "shadowOffset": undefined,
      "shadowOpacity": undefined,
      "shadowRadius": undefined,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      1
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 12,
          "textAlign": "center",
        }
      }
    >
      test
    </Text>
  </View>
</View>
`;

exports[`Rendering Render expansions Should render given expansions and highlight the selected one 1`] = `
Array [
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "zIndex": 1,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#f4f4f4",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 0,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 2,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={Object {}}
      >
        hello
      </Text>
    </View>
  </View>,
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "zIndex": 1,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#f4f4f4",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 0,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 2,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        world
      </Text>
    </View>
  </View>,
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "zIndex": 1,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#f4f4f4",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 0,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 2,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={Object {}}
      >
        again
      </Text>
    </View>
  </View>,
]
`;
